### DISCUSSION TITLE
Selecting compression per column?

### DISCUSSION THREAD

**Discussion by javafanboy** [jerolba/parquet-carpet#49] [Q&A]
I belive the Parquet spec allow for selecting compression per column (at least tools report it separatly per column giving me the impression it is suported) - is there a way to do this with Carpet?

I have noticed that even though the total size of the file becomes somewhat smaller many of the columns in my files are LARGER with compression than without and it would be very attractive to only enable compression for the ones where it actually provides an improvement as this would reduce total file size further and also improve both encoding and decoding performance...

**Comment by jerolba** [comment]
AFAIK, using standard Parquet Java API you can not select which compression codec to use in each column.

Carpet, via `CarpetWriter.Builder`, exposes all [`ParquetWriter.Builder`](https://github.com/apache/parquet-java/blob/master/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetWriter.java#L466)  methods that configure the internals of Parquet.
If you find in `ParquetWriter.Builder` a method that configures it and it is not exposed in `CarpetWriterBuilder` open an issue.

Probably reimplementing ParquetWriter and how columns are defined you can customize the compression algorithm per column.

I suggest to play with the configuration of dictionary encoding per column and test if the compression issue is related with it:

```java
var writer = new CarpetWriter.Builder<>(outputStream, YourRecord.class)
                    .withDictionaryEncoding("column_name", false)
                    .build();
```

**Comment by javafanboy** [comment]
Thanks for the quick reply - will try the dictionary stuff!
